lib/ddtrace/transport/traces.rb

```diff
@@ -8,25 +9,177 @@ module Traces
       class Parcel
         include Transport::Parcel
 
-        def count
-          data.length
+        attr_reader :trace_count
+
+        def initialize(data, trace_count)
```

Comment:
I don't think this makes sense with the current design, and altering this has a lot of side effects.

`Traces::Parcel` in the current design is supposed to be a protocol agnostic package of trace data, to be created by something that doesn't have knowledge of how the transport works. By requiring the `Parcel` to be given encoded data like this (along with its trace count), it implicitly requires a knowledge of the transport and its current API state to properly construct, rendering this an object with strong coupling to internal transport behavior.

That said, I think there's an argument to be made that we should change the design, and that to support encoding in chunks, it might also require some different kind of construct.

Short term, maybe we can remedy this by leaving the existing `Traces::Parcel` as is, but creating a new `Traces::EncodedParcel` which results from encoding traces from a parcel during chunking.

Long term, I think perhaps it shouldn't create parcels at all; the transport should only receive generic requests, and return generic responses... any trace specific behavior should be in some kind of `Traces::Transport` that wraps a generic transport (HTTP/IO/UDS, etc.) Then we shouldn't have a need for any parcels.

Reply:
The reason I changed this Parcel is because this Parcel is actually a subtype, a `Traces::Parcel`, it inherits behaviour from the generic parcel `Transport::Parcel`, which is agnostic.

I did have the additional information I needed (`trace_count`) under another carrier object, which I believe was `Traces::Request`, but the parcel seemed like a better carrier for it. To be fair, I'm not 100% sure on the role of Parcel after the changes, so I'm very much open to changing this design.